www/js/app.js

```diff
@@ -780,76 +780,86 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
-                    setContentInjectionMode('jquery');
-                });
-                return;
-            }
-            if (!isMessageChannelAvailable()) {
-                uiUtil.systemAlert('The MessageChannel API is not available on your device. Falling back to JQuery mode', 'MessageChannel API not available').then(function () {
-                    setContentInjectionMode('jquery');
-                });
-                return;
-            }
             var protocol = window.location.protocol;
-            if (!isServiceWorkerReady()) {
-                $('#serviceWorkerStatus').html("ServiceWorker API available : trying to register it...");
-                if (navigator.serviceWorker.controller) {
-                    console.log("Active service worker found, no need to register");
-                    serviceWorkerRegistration = true;
-                    // Remove any jQuery hooks from a previous jQuery session
-                    $('#articleContent').contents().remove();
-                    // Create the MessageChannel and send 'init'
-                    initOrKeepAliveServiceWorker();
-                    refreshAPIStatus();
-                } else {
-                    navigator.serviceWorker.register('../service-worker.js').then(function (reg) {
-                        // The ServiceWorker is registered
-                        serviceWorkerRegistration = reg;
-                        // We need to wait for the ServiceWorker to be activated
-                        // before sending the first init message
-                        var serviceWorker = reg.installing || reg.waiting || reg.active;
-                        serviceWorker.addEventListener('statechange', function(statechangeevent) {
-                            if (statechangeevent.target.state === 'activated') {
-                                // Remove any jQuery hooks from a previous jQuery session
-                                $('#articleContent').contents().remove();
-                                // Create the MessageChannel and send the 'init' message to the ServiceWorker
+            // Since Firefox 103, the ServiceWorker API is not available any more in Webextensions. See https://hg.mozilla.org/integration/autoland/rev/3a2907ad88e8 and https://bugzilla.mozilla.org/show_bug.cgi?id=1593931
+            // Previously, the API was available, but failed to register (which we could trap a fews lines below).
+            // So we now need to suggest a switch to the PWA if we are inside a Firefox Extension and the ServiceWorker API is unavailable.
+            // Even if some older firefox versions do not support ServiceWorkers at all (versions 42, 43, 45ESR, 52ESR, 60ESR and 68ESR, based on https://caniuse.com/serviceworkers). In this case, the PWA will not work either.
+            if (protocol === 'moz-extension:' && !isServiceWorkerAvailable()) {
+                launchMozillaExtensionServiceWorker();
+            }
+            else {
```

Comment:
Use same coding style as surrounding code 😉
```suggestion
            } else {
```